pantheon/src/main/java/tech/pegasys/pantheon/cli/PantheonCommand.java

```diff
@@ -785,24 +726,61 @@ private InetAddress autoDiscoverDefaultIP() {
     return autoDiscoveredDefaultIP;
   }
 
-  private EthNetworkConfig ethNetworkConfig() {
+  private EthNetworkConfig ethNetworkConfig(final NetworkName network) {
     final EthNetworkConfig predefinedNetworkConfig;
-    if (rinkeby) {
-      predefinedNetworkConfig = EthNetworkConfig.rinkeby();
-    } else if (ropsten) {
-      predefinedNetworkConfig = EthNetworkConfig.ropsten();
-    } else if (goerli) {
-      predefinedNetworkConfig = EthNetworkConfig.goerli();
-    } else {
-      predefinedNetworkConfig = EthNetworkConfig.mainnet();
+
+    switch (network) {
+      case ROPSTEN:
+        predefinedNetworkConfig = EthNetworkConfig.ropsten();
+        break;
+      case RINKEBY:
+        predefinedNetworkConfig = EthNetworkConfig.rinkeby();
+        break;
+      case OTTOMAN:
+        predefinedNetworkConfig = EthNetworkConfig.ottoman();
+        if (genesisFile() == null) {
+          throw new ParameterException(
+              new CommandLine(this),
+              "A genesis file must be defined with --private-genesis-file option to use Ottoman network.");
+        }
+        break;
+      case GOERLI:
+        predefinedNetworkConfig = EthNetworkConfig.goerli();
+        break;
+      case DEV:
+        predefinedNetworkConfig = EthNetworkConfig.dev();
+        break;
+      case MAINNET:
+      default:
+        predefinedNetworkConfig = EthNetworkConfig.mainnet();
+        break;
     }
+
     return updateNetworkConfig(predefinedNetworkConfig);
   }
 
   private EthNetworkConfig updateNetworkConfig(final EthNetworkConfig ethNetworkConfig) {
     final EthNetworkConfig.Builder builder = new EthNetworkConfig.Builder(ethNetworkConfig);
     if (genesisFile() != null) {
       builder.setGenesisConfig(genesisConfig());
+      // if we use a custom genesis, it means that we are not in any of the known networks and even
+      // if
+      // some properties of our ustom network are the same as one of the known networks, we can't
+      // connect to the same bootnodes or with the same network-id. We then ensure that these two
+      // options are set.
+      if (networkId == null && bootstrapNodes == null) {
```

Comment:
I'd agree with Danno's comment that this restriction seems too strict.  The current behaviour of the predefined network providing a base with the ability to override genesis, network id or boot nodes individually makes a lot of sense.  The recent Constantinople delay is one case where you may have wanted to override the genesis to remove the Constantinople block but not have to upgrade the whole of Pantheon (which often requires extra testing and approvals).

Reply:
Ok, I will try to change that.

Reply:
Changed. It was mostly due to the removal of network id option, so getting setting the option default to chain id enabled to have something smoother without the need to define options and still the ability to override.